src/test/groovy/com/synacy/lesson03/exercise/EnrollmentServiceTest.groovy

```diff
@@ -0,0 +1,127 @@
+package com.synacy.lesson03.exercise
+
+import com.synacy.lesson03.exercise.domain.CourseClass
+import com.synacy.lesson03.exercise.domain.EnrollmentDto
+import com.synacy.lesson03.exercise.domain.Student
+import com.synacy.lesson03.exercise.domain.StudentEmailType
+import com.synacy.lesson03.exercise.domain.StudentProfile
+import com.synacy.lesson03.exercise.domain.StudyLoad
+import spock.lang.Specification
+
+/**
+ * Created by michael on 5/4/17.
+ */
+class EnrollmentServiceTest extends Specification {
+
+    EnrollmentService service
+    StudentProfileService studentProfileService
+    CourseClassService courseClassService
+    EnrollmentNotificationService enrollmentNotificationService
+    StudyLoadFormatter studyLoadFormatter
+    SystemService systemService
+
+    void setup() {
+        service = new EnrollmentService()
+
+        studentProfileService = Mock()
+        courseClassService = Mock()
+        enrollmentNotificationService = Mock()
+        studyLoadFormatter = Mock()
+        systemService = Mock()
+
+        service.setStudentProfileService(studentProfileService)
+        service.setCourseClassService(courseClassService)
+        service.setEnrollmentNotificationService(enrollmentNotificationService)
+        service.setStudyLoadFormatter(studyLoadFormatter)
+        service.setSystemService(systemService)
+    }
+
+    def "processEnrollment should update student profile"() {
+        given:
+            EnrollmentDto enrollmentDetails = Mock()
+
+            StudentProfile studentProfile = Mock()
+
+            CourseClass class1 = Mock()
+            CourseClass class2 = Mock()
+
+            def enrolledClasses = [
+                class1, class2
+            ]
+
+            enrollmentDetails.getStudentProfile() >> studentProfile
+            enrollmentDetails.getEnrolledClasses() >> enrolledClasses
+
+        when:
+            service.processEnrollment(enrollmentDetails)
+
+        then:
+            1 * studentProfileService.updateStudentProfile(enrollmentDetails.getStudentProfile())
```

Comment:
a little improvement is to place the object `studentProfile` directly here than calling the method `enrollmentDetails.getStudentProfile()` again